metagraph/src/graph/representation/succinct/dbg_succinct_cached.hpp

```diff
@@ -0,0 +1,372 @@
+#ifndef __DBG_SUCCINCT_CACHED__
+#define __DBG_SUCCINCT_CACHED__
+
+#include "boss.hpp"
+#include "dbg_succinct.hpp"
+#include "graph/representation/base/dbg_wrapper.hpp"
+#include "kmer/kmer_extractor.hpp"
+#include "common/caches.hpp"
+
+namespace mtg {
+namespace graph {
+
+
+// A wrapper which caches computed node sequences for DBGSuccinct graphs.
+// This allows for faster get_node_sequence and call_incoming_kmers calls.
+class DBGSuccinctCached;
+
+std::shared_ptr<DBGSuccinctCached>
+make_cached_dbgsuccinct(std::shared_ptr<const DBGSuccinct> graph, size_t cache_size = 1024);
```

Comment:
Better make it a public method of `DBGSuccinct`: `get_cached_view(size_t cache_size)`.

Reply:
or `get_const_cached_view()`

Reply:
Sure, I can add `get_cached_view`, but I would still keep `make_cached_dbgsuccinct`. `get_cached_view` would have to be passed an alias of `DBGSuccinct`, so one would have to do extra work to keep that `DBGSuccinct` in scope once a cached graph is generated. This'll make it more annoying to implement `build_graph` from the unit tests.

Reply:
I think this should not be necessary. If this is only needed for unit tests, move it to there. Then you'll probably see how you can get rid of it completely.